modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/JdbcQueryExecuteResult.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.proto.query.event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.ignite.client.proto.ClientMessagePacker;
+import org.apache.ignite.client.proto.ClientMessageUnpacker;
+import org.apache.ignite.internal.tostring.S;
+
+/**
+ * JDBC query execute result.
+ */
+public class JdbcQueryExecuteResult extends JdbcResponse {
+    /** Query result rows. */
+    private List<JdbcQuerySingleResult> results;
+
+    /**
+     * Constructor. For deserialization purposes only.
+     */
+    public JdbcQueryExecuteResult() {
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param status Status code.
+     * @param err Error message.
+     */
+    public JdbcQueryExecuteResult(int status, String err) {
+        super(status, err);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param results Results.
+     */
+    public JdbcQueryExecuteResult(List<JdbcQuerySingleResult> results) {
+        super();
+
+        this.results = results;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void writeBinary(ClientMessagePacker packer) {
+        super.writeBinary(packer);
+
+        if (status() != STATUS_SUCCESS)
+            return;
```

Comment:
Let's avoid any logic from serialization.

If you want to have different messages to safe few bytes, then 
it would be better to create different Java classes representing these messages.

So, we won't care about binary message layout here.

Reply:
I got rid of the use of business logic fields here. Now I have a separate flag.